include/deal.II/distributed/tria_base.h

```diff
@@ -398,6 +398,455 @@ namespace parallel
       const typename dealii::Triangulation<dim, spacedim>::MeshSmoothing
                  smooth_grid = (dealii::Triangulation<dim, spacedim>::none),
       const bool check_for_distorted_cells = false);
+
+    /**
+     * Reset this triangulation into a virgin state by deleting all data.
+     *
+     * Note that this operation is only allowed if no subscriptions to this
+     * object exist any more, such as DoFHandler objects using it.
+     */
+    virtual void
+    clear() override;
+
+    using cell_iterator =
+      typename dealii::Triangulation<dim, spacedim>::cell_iterator;
+
+    using CellStatus =
+      typename dealii::Triangulation<dim, spacedim>::CellStatus;
+
+    /**
+     * Save the triangulation into the given file. This file needs to be
+     * reachable from all nodes in the computation on a shared network file
+     * system. See the SolutionTransfer class on how to store solution vectors
+     * into this file. Additional cell-based data can be saved using
+     * register_data_attach().
+     */
+    virtual void
+    save(const std::string &filename) const = 0;
+
+    /**
+     * Load the triangulation saved with save() back in. Cell-based data that
+     * was saved with register_data_attach() can be read in with
+     * notify_ready_to_unpack() after calling load().
+     */
+    virtual void
+    load(const std::string &filename, const bool autopartition = true) = 0;
+
+    /**
+     * Save additional cell-attached data into the given file. The first
+     * arguments are used to determine the offsets where to write buffers to.
+     *
+     * Called by @ref save.
+     */
+    void
+    save_attached_data(const unsigned int global_first_cell,
+                       const unsigned int global_num_cells,
+                       const std::string &filename) const;
```

Comment:
Does `save_attached_data` and `load_attached_data` need to be `public` when they are only called by `save` and `load`, respectively?

I would move them into the `protected` scope, or make them part of `CellAttachedData`.